runtime-sdk/modules/evm/src/lib.rs

```diff
@@ -356,9 +523,75 @@ impl module::MigrationHandler for Module {
     }
 }
 
-impl module::AuthHandler for Module {}
+impl<Cfg: Config> module::AuthHandler for Module<Cfg> {
+    fn authenticate_tx<C: Context>(ctx: &mut C, tx: &Transaction) -> Result<(), CoreError> {
+        // We're only interested in transactions that can be paid with tokens
+        // from the corresponding EVM account.
+        match tx.call.method.as_str() {
+            "evm.Create" | "evm.Call" | "evm.Withdraw" => {}
+            _ => return Err(CoreError::NotAuthenticated),
+        }
+
+        let params = Self::params(ctx.runtime_state());
+        let den = params.token_denomination;
+
+        let evm_acct_addr = Self::derive_caller_from_tx_auth_info(&tx.auth_info)
+            .map_err(|e| CoreError::MalformedTransaction(anyhow::Error::new(e)))?;
+
+        // Check and update nonces on all signer accounts.
+        let mut sdk_accounts_store =
+            storage::PrefixStore::new(ctx.runtime_state(), &modules::accounts::MODULE_NAME);
+        let mut sdk_accounts = storage::TypedStore::new(storage::PrefixStore::new(
```

Comment:
This seems to rely on internal implementation details of the accounts module which is bad. Instead, this should be exposed as an API on the accounts module and called via `Cfg::Accounts`.